course_catalog/etl/loaders.py

```diff
@@ -0,0 +1,45 @@
+"""Course catalog data loaders"""
+from django.contrib.contenttypes.models import ContentType
+
+from course_catalog.models import Course, Program, ProgramItem
+
+
+def load_course(course_data):
+    """Load the course into the database"""
+    course_id = course_data.pop("course_id")
+    course, _ = Course.objects.update_or_create(
+        course_id=course_id, defaults=course_data
+    )
+    return course
+
+
+def load_program(program_data):
+    """Load the program into the database"""
+    program_id = program_data.pop("program_id")
+    courses_data = program_data.pop("courses")
+
+    program, _ = Program.objects.update_or_create(
+        program_id=program_id, defaults=program_data
+    )
+
+    for position, course_data in enumerate(courses_data):
+        # skip courses that don't define a course_id
+        if not course_data.get("course_id", None):
+            continue
```

Comment:
Might end up with empty programs if none of the courses in the program have course ids. For instance, on my local instance and on RC, the "Analog Learning" and "Digital Learning" programs.  Not sure how likely that is on production.

Reply:
Yeah, that's definitely possible, but that amounts to a configuration mistake on MM. I'm inclined to leave it because it might be more obvious what is wrong if a program is empty rather than the program not showing up at all?

Reply:
👍